src/BaseITwinsApiClient.ts

```diff
@@ -0,0 +1,376 @@
+  protected static readonly ODataParamMapping: ParameterMapping<
+    Pick<ODataQueryParams, "search" | "skip" | "top">
+  > = {
+    top: "$top",
+    skip: "$skip",
+    search: "$search",
+  } as const;
+
+  /**
+   * Maps the properties some of the {@link ODataQueryParams} and all of the {@link ITwinsQueryArg} to their corresponding query parameter names.
+   *
+   * @remarks
+   * This mapping is used to translate internal property names to the expected parameter names
+   * when constructing iTwins queries. Properties mapped to empty strings are excluded from
+   * the query string as they should be sent as headers instead.
+   *
+   * The mapping includes both OData query parameters (prefixed with $) and iTwins-specific
+   * parameters for filtering and pagination.
+   *
+   * @readonly
+   */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  protected static readonly ITwinsGetQueryParamMapping: ParameterMapping<
+    ITwinsQueryArg & Pick<ODataQueryParams, "filter" | "orderby" | "select">
+  > = {
+    subClass: "subClass",
+    type: "type",
+    status: "status",
+    search: "$search",
+    displayName: "displayName",
+    // eslint-disable-next-line id-denylist
+    number: "number",
+    top: "$top",
+    skip: "$skip",
+    parentId: "parentId",
+    iTwinAccountId: "iTwinAccountId",
+    includeInactive: "includeInactive",
+    resultMode: "",
+    queryScope: "",
+    filter: "$filter",
+    orderby: "$orderby",
+    select: "$select",
+  } as const;
+
+  /**
+   * Maps the properties of class and subclass to their corresponding query parameter names.
+   *
+   * @remarks
+   * This mapping is used to translate internal property names to the expected parameter names
+   * when constructing repository queries.
+   *
+   * @readonly
+   */
+  protected static readonly repositoryParamMapping: ParameterMapping<{
+    class: Repository["class"];
+    subClass: Repository["subClass"];
+  }> = {
+    class: "class",
+    subClass: "subClass",
+  } as const;
+
+  /**
+   * The base URL for iTwins API endpoints.
+   * The URL can be customized via the constructor parameter or automatically
+   * modified based on the IMJS_URL_PREFIX environment variable for different
+   * deployment environments.
+   *
+   * @readonly
+   */
+  protected readonly _baseUrl: string = "https://api.bentley.com/itwins";
+
+  /**
+   * Creates a new BaseClient instance for iTwins API operations
+   * @param url - Optional custom base URL, defaults to production iTwins API URL
+   *
+   * @example
+   * ```typescript
+   * // Use default production URL
+   * const client = new BaseClient();
+   *
+   * // Use custom URL for development/testing
+   * const client = new ITwinsAccessClient("https://dev-api.bentley.com/itwins");
+   * ```
+   */
+  public constructor(url?: string) {
+    super();
+    if (url !== undefined) {
+      this._baseUrl = url;
+    } else {
+      const urlPrefix = process.env.IMJS_URL_PREFIX;
```

Comment:
We might want to think about using the `globalThis.IMJS_URL_PREFIX` instead of the process environment variable.

 There is a reason for this but I can't quite remember, but I think it's the preferred way of doing this now.